src/main/java/dev/waterdog/packs/types/ZipResourcePack.java

```diff
@@ -54,8 +55,10 @@ public InputStream getStream(Path path) throws IOException {
 
     @Override
     public void saveToCache() throws IOException {
-        InputStream inputStream = Files.newInputStream(this.packPath);
-        this.cachedPack = inputStream.readAllBytes();
+        try (InputStream inputStream = Files.newInputStream(this.packPath)) {
+            byte[] allBytes = new byte[inputStream.available()];
+            this.cachedPack = Arrays.copyOf(allBytes, inputStream.read(allBytes));
+        }
```

Comment:
This code reads all the available bytes from the inputstream the same result as Java 11´s readAllBytes